spec/models/allocation_spec.rb

```diff
@@ -271,4 +271,59 @@
       end
     end
   end
+
+  describe 'automate pushing the primary pom to ndelius' do
+    let(:prison) { build(:prison).code }
+
+    context 'when a new allocation is created and a POM is set' do
+      before do
+        expect(PrisonOffenderManagerService).to receive(:get_pom_name).with(nomis_staff_id).and_return ['Bill', 'Jones']
```

Comment:
The pom name is stored in the allocation, so we shouldn't need to fetch it again

Reply:
@starswan The only risk I see with using the 'cached' `pom_name` field is this:

When saved into the `pom_name` field of an Allocation record, the POM name is transformed into a single string.
e.g. API response `['Bill', 'Jones']` is saved as string `"Jones, Bill"`

But the Community API requires the POM name to be supplied as a pair of `[forename, surname]` – which fits perfectly with the response we receive from the `PrisonOffenderManagerService`.

To use the `pom_name` field value, we'd need to perform the opposite transformation to split the `"Surname, Forename"` string back into an array.

It's _probably safe_ to reverse that transformation _in most cases_. But I wonder if there will ever be a case where one of the names contain a comma (`,`) which breaks this reverse transformation.

e.g. given an original name:

```ruby
["John", "Jones,Smith"]
```

our `pom_name` field would have value:

```ruby
"Jones,Smith, John"
```

and we'd struggle to reverse the transformation.

It could end up as:

```ruby
["Smith, John", "Jones"]
```

I have no idea how likely this scenario is, or if it's even possible for a POM name to contain a comma. I just want to point out the risk of us attempting to un-transform the POM name field to fit our needs here.